tests/common/devices/eos.py

```diff
@@ -103,9 +103,61 @@ def no_shutdown_multiple(self, interfaces):
+                 link is admin down(cable not present):
+                     Ethernet1/1 is administratively down, line protocol is notpresent (disabled)
+                 link is admin down(cable present):
+                     Ethernet2/1 is administratively down, line protocol is down (disabled)
+                 link is admin up&oper up:
+                     Ethernet2/1 is up, line protocol is up (connected)
+                 link is admin up&oper down:
+                     Ethernet2/1 is down, line protocol is down (notconnect)
+        In conclusion, if 'up' found in output line, link is oper up&admin up.
+        Link could not be admin down&oper up.
+        This function could not tell if it's admin up&oper down
+        """
         show_int_result = self.eos_command(
             commands=['show interface %s' % interface_name])
-        return 'Up' in show_int_result['stdout_lines'][0]
+        return 'up' in show_int_result['stdout_lines'][0].lower()
+
+    def check_intf_link_oper_state(self, interface_name):
+        """
+        This function returns oper state for eos fanout
+        e.g. Et1/1 is admin up, oper down
+             Et2/1 is admin down
+             Et3/1 is oper up
+        Et1/1      str2-7804-lc7-1-Ethernet0            notconnect   1102     full   100G   100GBASE-CR4
+        Et2/1      str2-7804-lc7-1-Ethernet4            disabled     1103     full   100G   100GBASE-CR4
+        Et3/1      str2-7804-lc7-1-Ethernet8            connected    1104     full   100G   100GBASE-CR4
+        """
+        show_int_result = self.eos_command(commands=['show interface status'])
+        for output_line in show_int_result['stdout_lines'][0]:
+            fields = output_line.split(' ')
+            output_port = fields[0].replace('Et', 'Ethernet')
+            if interface_name == output_port:
+                # if link is not oper up, we consider it as oper down, it could be admin down as well
+                if 'connected' not in fields:
+                    logging.info("Interface {} is oper down on {}".format(output_port, self.hostname))
+                    return False
+        return True
+
+    def check_intf_link_admin_state(self, interface_name):
+        """
+        This function returns admin state for eos fanout
+        """
+        show_int_result = self.eos_command(commands=['show interface status'])
+        for output_line in show_int_result['stdout_lines'][0]:
+            fields = output_line.split(' ')
+            output_port = fields[0].replace('Et', 'Ethernet')
+            if interface_name == output_port:
+                # if link is not admin down, we consider it as admin up, however, it could be oper down(notconnect)
+                if 'disabled' in fields:
+                    logging.info("Interface {} is admin down on {}".format(output_port, self.hostname))
+                    return False
+        return True
```

Comment:
I would also caution against adding new code that tries to interpret or directly parse text output from eos commands. The approach in #9127 would be much preferred as the JSON output gives a well-defined model.

Reply:
I agree with you in the json way.
but I think it's necessary to avoid the confusion in the function name as well. if we check for 'connected' then it's checking link oper status only, this is another thing I would want to clarify in this PR

Reply:
Sounds good, I'll let others comment on the preferred naming here. Just wanted to give my 2c on using JSON output.